aea/cli/config.py

```diff
@@ -213,14 +209,25 @@ def _check_set_field_name(self) -> None:
         top_level_key = self.json_path[0]
 
         if self.component_id:
-            config = self.component_id.package_type.configuration_class()
+            _config = self.component_id.package_type.configuration_class()
         else:
-            config = self.agent_config
+            _config = type(self.agent_config)
 
-        if top_level_key not in config.FIELDS_ALLOWED_TO_UPDATE:
+        if top_level_key not in _config.FIELDS_ALLOWED_TO_UPDATE:
             raise click.ClickException(
                 f"Field `{top_level_key}` is not allowed to change!"
             )
+        if _config == SkillConfig:
+            if len(self.json_path) < 3:
+                raise click.ClickException(
+                    f"Path {self.json_path} not valid for skill."
+                )
+            second_level_key = self.json_path[1]
+            third_level_key = self.json_path[2]
+            if third_level_key not in SkillConfig.NESTED_FIELDS_ALLOWED_TO_UPDATE:
+                raise click.ClickException(
+                    f"Field `{top_level_key}.{second_level_key}.{third_level_key}` is not allowed to change!"
+                )
```

Comment:
This adds the necessary validation for the skills...adhoc we can eventually improve